src/controllers/csi/gc/unmounted_test.go

```diff
@@ -0,0 +1,176 @@
+
+		require.NoError(t, err)
+		require.Len(t, unmountedVolumes, 3)
+		assert.Equal(t, testVersion1, unmountedVolumes[0].Name())
+		assert.Equal(t, testVersion2, unmountedVolumes[1].Name())
+		assert.Equal(t, testVersion3, unmountedVolumes[2].Name())
+	})
+	t.Run("multiple unmounted volumes are collected while mounted volume is present", func(t *testing.T) {
+		gc := NewMockGarbageCollector()
+		gc.mockMountedVolumeIDPath(testVersion3)
+		gc.mockUnmountedVolumeIDPath(testVersion1, testVersion2)
+
+		unmountedVolumes, err := gc.getUnmountedVolumes(testTenantUUID)
+
+		assert.NoError(t, err)
+		require.Len(t, unmountedVolumes, 2)
+		assert.Equal(t, testVersion1, unmountedVolumes[0].Name())
+		assert.Equal(t, testVersion2, unmountedVolumes[1].Name())
+	})
+}
+
+func TestIsUnmountedVolumeTooOld(t *testing.T) {
+	t.Run("default is false for current timestamp", func(t *testing.T) {
+		gc := CSIGarbageCollector{
+			maxUnmountedVolumeAge: defaultMaxUnmountedCsiVolumeAge,
+		}
+
+		isOlder := gc.isUnmountedVolumeTooOld(time.Now())
+
+		assert.False(t, isOlder)
+	})
+
+	t.Run("default is true for timestamp 14 days in past", func(t *testing.T) {
+		gc := CSIGarbageCollector{
+			maxUnmountedVolumeAge: defaultMaxUnmountedCsiVolumeAge,
+		}
+
+		isOlder := gc.isUnmountedVolumeTooOld(time.Now().AddDate(0, 0, -15))
+
+		assert.True(t, isOlder)
+	})
+
+	t.Run("is true if maxUnmountedCsiVolumeAge == 0", func(t *testing.T) {
+		gc := CSIGarbageCollector{
+			maxUnmountedVolumeAge: 0,
+		}
+
+		isOlder := gc.isUnmountedVolumeTooOld(time.Now().AddDate(0, 0, 15))
+
+		assert.True(t, isOlder)
+	})
+}
+
+func TestRemoveUnmountedVolumesIfNecessary(t *testing.T) {
+
+	t.Run("remove only too old unmounted volume", func(t *testing.T) {
+		gc := NewMockGarbageCollector()
+		gc.mockUnmountedVolumeIDPath(testVersion1, testVersion2, testVersion3)
+
+		unmountedVolumes, err := gc.getUnmountedVolumes(testTenantUUID)
+		require.NoError(t, err)
+		require.NotNil(t, unmountedVolumes)
+
+		oldVolume := unmountedVolumes[0]
+		err = gc.fs.Chtimes(filepath.Join(testVolumeFolderPath, oldVolume.Name()), oldVolume.ModTime(), oldVolume.ModTime().AddDate(0, 0, -15))
+		require.NoError(t, err)
+
+		older := gc.isUnmountedVolumeTooOld(oldVolume.ModTime())
+		require.True(t, older)
+
+		gc.removeUnmountedVolumesIfNecessary(unmountedVolumes, testTenantUUID)
+		oldVolumeExists, err := afero.DirExists(gc.fs, filepath.Join(testVolumeFolderPath, oldVolume.Name()))
+		require.NoError(t, err)
+		assert.False(t, oldVolumeExists)
+
+		for _, remainingVolume := range unmountedVolumes[1:] {
+			volumeExists, err := afero.DirExists(gc.fs, filepath.Join(testVolumeFolderPath, remainingVolume.Name()))
+			require.NoError(t, err)
+			assert.True(t, volumeExists)
+		}
+	})
+
+}
+
+func TestDetermineMaxUnmountedVolumeAge(t *testing.T) {
+
+	t.Run("no env set ==> use default", func(t *testing.T) {
+		os.Setenv(maxUnmountedCsiVolumeAgeEnv, "")
+		defer os.Unsetenv(maxUnmountedCsiVolumeAgeEnv)
```

Comment:
I've overlooked that in my earlier review, please refactor setting the env var in the constructor so it can be mocked. having to set and unset env vars in a unit test doesn't sit right with me